src/server.rs

```diff
@@ -721,6 +721,31 @@ impl LanguageServer for LspServer {
         };
         Ok(Some(response))
     }
+    async fn references(
+        &self,
+        params: lsp::ReferenceParams,
+    ) -> Result<Option<Vec<lsp::Location>>> {
+        let key =
+            params.text_document_position.text_document.uri.clone();
+        let store = self.store.lock().unwrap();
+        let contents = match store.get(&key) {
+            Some(v) => v,
+            None => {
+                error!(
+                    "textDocument/references called on unknown file {}",
+                    key
+                );
+                return Err(lspower::jsonrpc::Error::invalid_params(
+                    format!("file not opened: {}", key),
+                ));
+            }
+        };
+        let pos = params.text_document_position.position;
+        let pkg = parse_and_analyze(contents);
+        let node = find_node(walk::Node::Package(&pkg), pos);
+
+        Ok(Some(find_references(key.clone(), node)))
```

Comment:
Does clippy complain about this? I wouldn't think `key` would need to be cloned here, since it's the last use.

Reply:
Clippy did not complain about this, but I'll change it anyway. Thanks for catching it.

As an aside, do we need to pass `--features lsp2` to clippy when running it in CI? I get different results on my machine when running it with/without that flag. I can add it now if want to do that.